journalalerts/journalalerts.go

```diff
@@ -0,0 +1,404 @@
+		return aoserrors.Wrap(err)
+	}
+
+	if err = instance.journal.AddMatch("_SYSTEMD_UNIT=init.scope"); err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	if err = instance.journal.SeekTail(); err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	if _, err = instance.journal.Previous(); err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	cursor, err := instance.cursorStorage.GetJournalCursor()
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	if cursor != "" {
+		if err = instance.journal.SeekCursor(cursor); err != nil {
+			return aoserrors.Wrap(err)
+		}
+
+		if _, err = instance.journal.Next(); err != nil {
+			return aoserrors.Wrap(err)
+		}
+	}
+
+	go instance.handleChannels()
+
+	return nil
+}
+
+func (instance *JournalAlerts) handleChannels() {
+	result := sdjournal.SD_JOURNAL_APPEND
+	journalTicker := time.NewTicker(journalSavePeriod)
+
+	ctx, cancelFunction := context.WithCancel(context.Background())
+
+	instance.journalCancelFunction = cancelFunction
+
+	for {
+		select {
+		case <-journalTicker.C:
+			if err := instance.storeCurrentCursor(); err != nil {
+				log.Error("Can't store journal cursor: ", err)
+			}
+
+		case <-ctx.Done():
+			journalTicker.Stop()
+
+		default:
+			if result != sdjournal.SD_JOURNAL_NOP {
+				if err := instance.processJournal(); err != nil {
+					log.Errorf("Journal process error: %s", err)
+				}
+			}
+
+			if result = instance.journal.Wait(waitJournalTimeout); result < 0 {
+				log.Errorf("Wait journal error: %s", syscall.Errno(-result))
+			}
+		}
+	}
+}
+
+func (instance *JournalAlerts) processJournal() (err error) {
+	for {
+		count, err := instance.journal.Next()
+		if err != nil {
+			return aoserrors.Wrap(err)
+		}
+
+		if count == 0 {
+			return nil
+		}
+
+		entry, err := instance.journal.GetEntry()
+		if err != nil {
+			return aoserrors.Wrap(err)
+		}
+
+		if entry == nil {
+			return nil
+		}
+
+		unit := entry.Fields[sdjournal.SD_JOURNAL_FIELD_SYSTEMD_UNIT]
+
+		if unit == "init.scope" {
+			if priority, err := strconv.Atoi(entry.Fields[sdjournal.SD_JOURNAL_FIELD_PRIORITY]); err != nil ||
+				priority > instance.config.ServiceAlertPriority {
+				continue
+			}
+
+			unit = entry.Fields["UNIT"]
+		}
+
+		// with cgroup v2 logs from container do not contains _SYSTEMD_UNIT due to restrictions
+		// that's why id should be extracted from _SYSTEMD_CGROUP
+		// format: /system.slice/system-aos@service.slice/AOS_INSTANCE_ID
+		if len(unit) == 0 {
+			systemdCgroup := entry.Fields[sdjournal.SD_JOURNAL_FIELD_SYSTEMD_CGROUP]
+
+			if len(systemdCgroup) == 0 {
+				continue
+			}
+
+			// add prefix 'aos-service@' and postfix '.service'
+			// to service uuid and get proper seervice object from DB
+			unit = systemdCgroup
+		}
+
+		alert := cloudprotocol.AlertItem{
+			Timestamp: time.Unix(int64(entry.RealtimeTimestamp/microSecondsInSecond),
+				int64((entry.RealtimeTimestamp%microSecondsInSecond)*1000)),
+		}
+
+		instance.fillServiceInstanceAlert(&alert, entry, unit)
+
+		if alert.Payload == nil {
```

Comment:
It is not clear that fill.. function change alert.Payload inside. You should look inside function to understand what it does. I suggest the following:
```go
if alert.Payload = getServiceInstanceAlert(...); alert.payload != nil {
    alert.Tag = cloudprotocol.AlertTagServiceInstance
} else if alert.Payload = get...; alert.Payload != nil {
    alert.Tag = ...
} else {
    continue
}

instance.sender.SendAlert(alert)
```

get... function return nil or found alert type instance.